gwpy/timeseries/timeseries.py

```diff
@@ -1924,28 +1935,33 @@ def q_transform(self, qrange=(4, 64), frange=(0, numpy.inf),
         ny = frequencies.size
         out = Spectrogram(numpy.zeros((nx, ny)), x0=outseg[0], dx=tres,
                           frequencies=frequencies)
-        # FIXME: bug in Array2D.yindex setting
-        out._yindex = type(out.y0)(frequencies, out.y0.unit)
         # record Q in output
         out.q = peakq
         # interpolate rows
+        xout = numpy.arange(outseg[0], outseg[1], tres)
         for i, row in enumerate(norms):
-            interp = InterpolatedUnivariateSpline(
-                row.times.value, row.value)
-            out[:, i] = interp(out.times.value)
+            xrow = numpy.arange(row.x0.value, (row.x0 + row.duration).value,
+                                row.dx.value)
+            interp = InterpolatedUnivariateSpline(xrow, row.value)
+            out[:, i] = interp(xout)
+
+        if fres is None:
+            return out
 
         # then interpolate the spectrogram to increase the frequency resolution
         # --- this is done because duncan doesn't like interpolated images
         #     because they don't support log scaling
-        if fres is None:  # unless user tells us not to
-            return out
-
-        interp = interp2d(out.times.value, frequencies, out.value.T,
-                          kind='cubic')
-        freqs2 = numpy.arange(planes.frange[0], planes.frange[1], fres)
-        new = Spectrogram(interp(out.times.value, freqs2 + fres/2.).T,
-                          x0=outseg[0], dx=tres,
-                          f0=planes.frange[0], df=fres)
+        interp = interp2d(xout, frequencies, out.value.T, kind='cubic')
+        if not logf:
+            outfreq = numpy.arange(planes.frange[0], planes.frange[1], fres)
+        else:
+            # using `~numpy.logspace` here to support numpy-1.1.7 for EPEL7,
```

Comment:
This should really say `numpy-1.7.1` but feel free to leave it, the comment serves its purpose.

Reply:
@duncanmmacleod, nice catch! It was simple enough to change real fast, so I did that.